more_itertools/tests/test_more.py

```diff
@@ -1109,3 +1095,21 @@ def key(x):
         actual = groupby_transform(iterable, key) # default valuefunc
         expected = groupby(iterable, key)
         self.assertAllGroupsEqual(actual, expected)
+
+
+class ContextTestCase(TestCase):
+    def test_basic(self):
+        before = []
+        after = []
+
+        @contextmanager
+        def managed():
+            before.append(1)
```

Comment:
This looks correct to me, but it would be more resistant to mistakes if we chose a different value (easiest to read if it were a meaningful string) for each behavior.